ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -483,6 +486,87 @@ public void delete(MessageWriteSet messageSetToDelete) throws StoreException {
     }
   }
 
+  @Override
+  public void updateTtl(MessageWriteSet messageSetToUpdate) throws StoreException {
+    checkStarted();
+    final Timer.Context context = metrics.ttlUpdateResponse.time();
+    try {
+      List<MessageInfo> infoList = messageSetToUpdate.getMessageSetInfo();
+      Offset indexEndOffsetBeforeCheck = index.getCurrentEndOffset();
+      for (MessageInfo info : infoList) {
+        IndexValue value = index.findKey(info.getStoreKey());
+        if (value == null) {
+          throw new StoreException("Cannot update TTL of " + info.getStoreKey() + " since it's not in the index",
```

Comment:
Should this fail the whole method, or can we print a warning and handle the rest of the blobs?

Reply:
Good question - generally we have followed the pattern of failing the whole request (look at put, get and delete) but its something we may be able to change in the future

Part of the problem comes from the fact that there is a single "stream" for all of these puts/deletes/updates. So if you don't like something in the middle, you'd essentially have to keep track of what you don't like and skip over the stream for the corresponding size. It's possible but we just haven't seen a practical reason to do so.

Also, the batch version of the API is only used by replication (the frontend always sends only one put/get/delete/update at a time) and current replication code would have problems if we silently skipped stuff in the middle because it wouldn't know how much to advance the token to.